app/rtc_compare.py

```diff
@@ -0,0 +1,483 @@
+                      f'Tolerance = {RTC_S1_PRODUCTS_ERROR_TOLERANCE}')
+                print(f'    - 1st value: {val_1}')
+                print(f'    - 2nd value: {val_2}\n')
+            return return_val
+
+        # Not a numerical array
+        return_val = np.array_equal(val_1, val_2)
+        if return_val:
+            if print_passed_element:
+                print('\033[32mPASSED.\033[00m', str_message_data_location)
+        else:
+            print('\033[91mFAILED.\033[00m', str_message_data_location)
+            print( '    - non-numerical scalar. Failed to pass the test.')
+            print(f'    - 1st value: {val_1}')
+            print(f'    - 2nd value: {val_2}\n')
+        return return_val
+
+    if len(shape_val_1) == 1:
+        # 1d vector
+
+        if issubclass(val_1.dtype.type, np.number):
+            # val_1 and val_2 are numeric numpy array
+            return_val = np.allclose(val_1, val_2, RTC_S1_PRODUCTS_ERROR_TOLERANCE, equal_nan=True)
+
+            if return_val:
+                if print_passed_element:
+                    print('\033[32mPASSED.\033[00m', str_message_data_location)
+            else:
+                print('\033[91mFAILED.\033[00m', str_message_data_location)
+                print('    - Numerical 1D array. Failed to pass the test. '
+                     f'Tolerance = {RTC_S1_PRODUCTS_ERROR_TOLERANCE}')
+                print_data_difference(val_1, val_2)
+            return return_val
+
+        # All other non-numerical cases, including the npy array with bytes
+        return_val = np.array_equal(val_1, val_2)
+        if return_val:
+            if print_passed_element:
+                print('\033[32mPASSED.\033[00m', str_message_data_location)
+        else:
+            print('\033[91mFAILED.\033[00m', str_message_data_location)
+            print('    non-numerical 1D array. Failed to pass the test.')
+        return return_val
+
+
+    if len(shape_val_1) >= 2:
+        return_val = np.allclose(val_1,
+                             val_2,
+                             RTC_S1_PRODUCTS_ERROR_TOLERANCE,
+                             equal_nan=True)
+        if return_val:
+            if print_passed_element:
+                print('\033[32mPASSED.\033[00m', str_message_data_location)
+        else:
+            print('\033[91mFAILED.\033[00m', str_message_data_location)
+            print(f'    {len(shape_val_1)}D raster array. Failed to pass the test. '
+                  f'Tolerance = {RTC_S1_PRODUCTS_ERROR_TOLERANCE}')
+            print_data_difference(val_1, val_2)
+        return return_val
+
+    # If the processing has reached here, that means
+    # val_1 and val_2 cannot be compated due to their shape difference
+    print('    - Detected an issue on the dataset shapes: ',
+            f'Dataset key: {str_key}, '
+            'dataset shape in the 1st HDF5: ', shape_val_1,
+            'dataset shape in the 2nd HDF5: ', shape_val_2)
+    return False
```

Comment:
This should raise an exception, right? Because you are not checking if the contents are the same or not.

Reply:
I agree on raising an error there. That part was to check the data shape, but we are already checking that earlier in the function as we were revising the code. So if the process has reached that part, that means we cannot compare `val_1` and `val_2` because of the reason we did not catch.